File: src/app/service/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = 'http://localhost:8080';

@Injectable({
  providedIn: 'root'
})
export class CommentService {

  //comment ở ViewPost

  constructor(private http:HttpClient) { }

  createComment(postId:number, postedBy: string, content:string):Observable<any>{
     const params = {
      postId: postId,
      postedBy : postedBy
     }

     return this.http.post<any>(`${BASIC_URL}/api/comments/create` ,content , {params})
  }

  getAllCommentsByPost(postId:number): Observable<any>{
    return this.http.get<any>(`${BASIC_URL}/api/comments/${postId}`)
  }

 
}

File: blogAngular/src/main/java/com/example/blogAngular/service/CommentService.java
package com.example.blogAngular.service;

import com.example.blogAngular.entity.Comment;

import java.util.List;

public interface CommentService {

    Comment createComment(Long postId, String postedBy, String content);

    List<Comment> getCommentsByPostId(Long postId);
    //void deleteComment(Long postId, Long commentId);
}
